File: -Parallel-computing-processes-Lr_1-CSharp-main/Lr_1/StopThread.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace Lr_1
{
    public class StopThread
    {
        private bool canStop = false;

        public void Run()
        {
            try
            {
                Thread.Sleep(3 * 1000);
            }
            catch(ThreadInterruptedException e)
            {
                Console.WriteLine(e.StackTrace);
            }
            canStop = true;
        }
        public bool IsCanStop()
        {
            return canStop;
        }
    }
}

File: -Parallel-computing-processes-Lr_1-Java-main/src/StopThread.java
public class StopThread implements Runnable {
    private boolean canStop=false;
    @Override
    public void run() {
        try {
            Thread.sleep(3*1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        canStop= true;
    }

    synchronized public boolean isCanStop()
    {
        return  canStop;
    }
}
